### ISSUE TITLE
Unstable

### ISSUE THREAD

**Issue by week3** [siggame/MegaMinerAI-Game-Ideas#68]
### **The Game:**
TLDR: Resource gathering units explode when they die. Try to take out enemy worker units in chain reactions.

### **Story**
Rival mining companies in an unregulated part of the galaxy. Sending robots to extract highly unstable fuel from the ground. One manager from each company commands the robots.

### **To Win**
Eliminate the competition by killing the opposing site manager. Tiebreaker have more value than your opponent (stored resources+units resource cost).

### **Gameplay**
Idea was taken from an age of empires mod where villagers explode on death
**https://www.youtube.com/watch?v=p-bcHocP_Io** 

### **Units**
**Mining bot**
Can gather resources and take them back to the main base. These bots are highly unstable due to their fuel source and if they die they explode. The site manager can self destruct their own robots at will. Bots packed to closely can cause chain reactions.

**Manager**
This is you. If they die its game over. They are on the map and can move like normal units (may a bit slower). 

**Ranged Unit** (come up with a fitting unit name for the story)
Can attack mining bots at a range outside the explosion distance. Do not explode when killed.

**Rover** (vehicle)
X amount of units can go inside the rover. The rover can transport other units inside of it. If the rover dies with units inside of it then the units die with it. Ranged units do little damage to the rover but mining bots have an easy time blowing it up.

**Basic unit counters**
Ranged units counter Mining Bots
Rovers counter Ranged units
Mining bots counter Rovers
Manager tries to survive

### Map
Grid based
Resources litter the map and prevent movement creating choke points.
Explosions destroy resource locations and can clear paths through them.
Resources are finite and disappear from a tile when a certain amount is taken.

### Some ideas to workshop
Some ideas that could be added to the game.

**Units spawn from command center**
Each player spawns with a fixed indestructible spawn point. Unit production is done here. Resources must be taken back here. No unit can go inside the spawn point.

or 

**Manager acts as spawn point**
Resources must be taken back to the manager. They can then create units. This gives the player a balance of keeping the Manager near miners were resource collection would happen the fastest. And spacing the manager outside of chain reaction zones.

**Alternate win condition**
There is an object in the center of the map surrounded by a ton of resources. Tunnel your way to it and take it back to your command post then defend it for x time to win.

**Walls**
Mining units can dump the resources they gathered to produce a barrier.

### Anti snowballing
The more units you have gathering resources the more you are risking to be taken out by a single chain reaction.

**Comment by TwinNumberOne** [comment]
my one worry is that when playing it the people programming the AI would just make it make sure the robots are far enough apart that they don't chain reaction explode or that they would then start trying to do a chain reaction explosion on the enemy using there own miner bot. Tldr most of the AI will be trying to do a chain reaction on the enemy and it would lead to a flat game.
